### PR TITLE
SAR-9480 Update identifier match storage and audit values

### PR DESCRIPTION
**Pull request by woollett-r**

### CODE REVIEW

**Review by robertom-binqua — CHANGES_REQUESTED** [review]

**Comment by robertom-binqua** [comment]
Richard are not compulsory things ... if you agree do it otherwise I am going to approve it.
Let me know

File: app/uk/gov/hmrc/incorporatedentityidentificationfrontend/models/IncorporatedEntityInformation.scala [hunk 1]
```diff
@@ -55,7 +57,7 @@ object IncorporatedEntityInformation {
     override def writes(incorporatedEntityInformation: IncorporatedEntityInformation): JsObject =
       Json.obj(
         companyProfileKey -> incorporatedEntityInformation.companyProfile,
-        identifiersMatchKey -> incorporatedEntityInformation.identifiersMatch,
+        identifiersMatchKey -> incorporatedEntityInformation.identifiersMatch.toString,
```
[the remark this answers is not in the document]
Comment (woollett-r) [reply app/uk/gov/hmrc/incorporatedentityidentificationfrontend/models/IncorporatedEntityInformation.scala:60]: Actually, I think it is the other way around. The "writes" method should serialise an instance of IncorporatedEntityInformation" and, therefore, should output a string corresponding to the appropriate validation response (or match).

File: app/uk/gov/hmrc/incorporatedentityidentificationfrontend/models/IncorporatedEntityInformation.scala [hunk 2]
```diff
@@ -72,6 +74,8 @@ object IncorporatedEntityInformation {
           case Some(businessVerification) => Json.obj(businessVerificationKey -> businessVerification)
           case None => Json.obj()
         }
+      } ++ {
+        Json.obj(IdentifiersMatchKey -> incorporatedEntityInformation.identifiersMatch.toString.contains(DetailsMatchedKey))
```
[the remark this answers is not in the document]
Comment (woollett-r) [reply app/uk/gov/hmrc/incorporatedentityidentificationfrontend/models/IncorporatedEntityInformation.scala:78]: This is incorrect (oops) as the match status is being converted to a boolean. This is only required when returning the journey data to the calling service.

File: app/uk/gov/hmrc/incorporatedentityidentificationfrontend/services/AuditService.scala [hunk 3]
```diff
@@ -64,13 +64,20 @@ class AuditService @Inject()(auditConnector: AuditConnector,
...
+    }
+
     journeyConfig.businessEntity match {
       case LimitedCompany =>
         val auditJson = Json.obj(
           "callingService" -> JsString(journeyConfig.pageConfig.optServiceName.getOrElse(appConfig.defaultServiceName)),
           "businessType" -> "UK Company",
           "companyNumber" -> companyNumber,
-          "isMatch" -> optIdentifiersMatch,
+          "isMatch" -> identifiersMatchStatus,
```
[the remark this answers is not in the document]
Comment (woollett-r) [reply app/uk/gov/hmrc/incorporatedentityidentificationfrontend/services/AuditService.scala:80]: As we define audit values for isMatch, VerificationStatus and RegisterApiStatus even if the source value is undefined I think it is better not to use a the "Block" approach for these fields. The only field that requires this approach is the CtUtr as this is optional for Registered Societies.

File: app/uk/gov/hmrc/incorporatedentityidentificationfrontend/models/IncorporatedEntityInformation.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (robertom-binqua) [reply app/uk/gov/hmrc/incorporatedentityidentificationfrontend/models/IncorporatedEntityInformation.scala:78]: okay

File: app/uk/gov/hmrc/incorporatedentityidentificationfrontend/services/CtEnrolmentService.scala [hunk 4]
```diff
@@ -39,13 +38,13 @@ class CtEnrolmentService @Inject()(storageService: StorageService,
             validateIncorporatedEntityDetailsService.validateIncorporatedEntityDetails(companyProfile.companyNumber, Some(ctutr)).flatMap {
               case DetailsMatched if journeyConfig.businessVerificationCheck =>
                 for {
-                  _ <- storageService.storeIdentifiersMatch(journeyId, identifiersMatch = true)
+                  _ <- storageService.storeIdentifiersMatch(journeyId, DetailsMatched)
```
[the remark this answers is not in the document]
Comment (robertom-binqua) [reply app/uk/gov/hmrc/incorporatedentityidentificationfrontend/services/CtEnrolmentService.scala]: Perfect

**Review by robertom-binqua — APPROVED** [review]

**Pull request merged**